internal/ethapi/api.go

```diff
@@ -1467,19 +1469,19 @@ func (r *RollupTransaction) toTransaction(txNonce uint64) *types.Transaction {
 	var tx *types.Transaction
 	c, _ := r.Calldata.MarshalText()
 	if r.Target == nil {
-		tx = types.NewContractCreation(txNonce, big.NewInt(0), uint64(*r.GasLimit), big.NewInt(0), c, r.Sender)
+		tx = types.NewContractCreation(txNonce, big.NewInt(0), uint64(*r.GasLimit), big.NewInt(0), c, r.Sender, r.L1RollupTxId)
 	} else {
-		tx = types.NewTransaction(txNonce, *r.Target, big.NewInt(0), uint64(*r.GasLimit), big.NewInt(0), c, r.Sender)
+		tx = types.NewTransaction(txNonce, *r.Target, big.NewInt(0), uint64(*r.GasLimit), big.NewInt(0), c, r.Sender, r.L1RollupTxId)
 	}
 	tx.AddNonceToWrappedTransaction(uint64(*r.Nonce))
 	return tx
 }
 
 // SendTxArgs represents the arguments to sumbit a new transaction into the transaction pool.
-type BlockBatches struct {
-	Timestamp   *hexutil.Uint64        `json:"timestamp"`
-	BlockNumber *hexutil.Uint64        `json:"blockNumber"`
-	Batches     [][]*RollupTransaction `json:"batches"`
+type GethSubmission struct {
```

Comment:
Am I right that the L1RollupTxId is what allows us to tie L1->L2 txs for instance to the outputs of Geth? And we need it because the tx hash isn't valid if the tx that gets processed can't be decoded?

OK yeah that makes sense and definitely LGTM!

Reply:
Yes on the first part, and yeah, that's one of the reasons on the second part. We need a unique identifier for transactions we pull from L1 queues, and unfortunately tx hash can't be. Like you said, someone could submit an incorrectly formatted and therefore invalid transaction, someone could submit the exact same transaction twice, etc. 

As a nice plus, presence or absence of this field in an L2 Tx Output lets us know with certainty where the tx originated (in an L1 queue or from a tx that has never been posted to L1, respectively)